web/modules/custom/registration_form_custom/registration_form_custom.module

```diff
@@ -0,0 +1,21 @@
+<?php
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\registration_form_custom\Form\NewUserRegisterForm;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\weather_block\Services\UserWeatherHandler;
+
+/**
+ * Implements hook_entity_insert().
+ */
+function registration_form_custom_entity_insert(\Drupal\user\Entity\User $entity) {
+  $user_id = $entity->id();
+  $city_value = \Drupal::request()->request->get('city');
+  $country_value = \Drupal::request()->request->get('country');
+  $interested_value = \Drupal::request()->request->get('interested');
```

Comment:
While form API will clear requests from values if validation fails, it is still not secure to use them in this way.